crates/wasmtime/src/types.rs

```diff
@@ -106,25 +106,7 @@ impl ValType {
             ValType::F32 => Some(ir::types::F32),
             ValType::F64 => Some(ir::types::F64),
             ValType::V128 => Some(ir::types::I8X16),
-            #[cfg(target_pointer_width = "64")]
-            ValType::ExternRef => Some(ir::types::R64),
-            #[cfg(target_pointer_width = "32")]
-            ValType::ExternRef => Some(ir::types::R32),
-            _ => None,
-        }
-    }
-
-    pub(crate) fn from_wasmtime_type(ty: ir::Type) -> Option<ValType> {
-        match ty {
-            ir::types::I32 => Some(ValType::I32),
-            ir::types::I64 => Some(ValType::I64),
-            ir::types::F32 => Some(ValType::F32),
-            ir::types::F64 => Some(ValType::F64),
-            ir::types::I8X16 => Some(ValType::V128),
-            #[cfg(target_pointer_width = "64")]
-            ir::types::R64 => Some(ValType::ExternRef),
-            #[cfg(target_pointer_width = "32")]
-            ir::types::R32 => Some(ValType::ExternRef),
+            ValType::ExternRef => Some(wasmtime_runtime::ref_type()),
```

Comment:
Could this perhaps go through the `Store` instead? One day if we ever support cross-compilation this'd be through that instead of `wasmtime-runtime`

Reply:
`wasmtime-environ` is the thing that should handle the cross compilation as it is about the environment of the compilation target, while `wasmtime-runtime` is always compiled into the target. So if we put it in `wasmtime-runtime`, then it is always going to reflect the compilation target.

If we wanted to support cross compilation at the `wasmtime` API level, then we would have to stop linking `wasmtime-runtime` directly, which would force us to (correctly) revisit calls to this function from the `wasmtime` crate. So I think we should be good here.

Reply:
It's true yeah we're a *long* way away from wasmtime supporting AOT compilation, but I just figured this'd be a relatively easy future-proofing.

Reply:
I think we are as future proofed as it makes sense to be given that this function is safely inside `wasmtime-runtime`, as mentioned above.

I think our best path forward for AOT cross compilation is a `wasmtime-aot` crate.